iceoryx_hoofs/test/moduletests/test_posix_periodic_timer.cpp

```diff
@@ -0,0 +1,222 @@
+// Copyright (c) 2021 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_hoofs/error_handling/error_handling.hpp"
+#include "iceoryx_hoofs/internal/units/duration.hpp"
+#include "iceoryx_hoofs/posix_wrapper/periodic_timer.hpp"
+#include "iceoryx_hoofs/testing/timing_test.hpp"
+#include "test.hpp"
+
+#include <chrono>
+#include <thread>
+
+using namespace ::testing;
+
+using namespace iox::units;
+using namespace iox::units::duration_literals;
+
+using Timer = iox::posix::PeriodicTimer;
```

Comment:
Could you please wrap the whole test in an anonymous namespace like [here](https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/aspice_swe3_4/example/iceoryx_component/test/moduletests/test_example_base_class.cpp#L23)? We had many troubles with clashing symbols leading to weird errors lately.